### PR TITLE
Write doc on new external feed block feature where decision is taken …

### PR DESCRIPTION
**Pull request by geirst**
…by the cluster controller.

@vekterli please review

### CODE REVIEW

File: en/reads-and-writes.html [hunk 1]
```diff
@@ -214,39 +214,58 @@ <h2 id="api-and-utilities">API and utilities</h2>
...
-    <td>disk or memory</td>
-  </tr><tr>
-    <th>content.proton.documentdb.attribute.resource_usage.feeding_blocked</th>
-    <td><em>Attribute Multivalue Mapping</em> or <em>Attribute Enum Store</em></td>
+    <th>cluster-controller.resource_usage.nodes_above_limit</th>
+    <td>The number of content nodes that are above one or more resource limits. When above 0, feeding is blocked.</td>
+  </tr>
+  <tr>
+    <th>content.proton.resource_usage.disk</th>
+    <td>A number between 0 and 1, indicating how huch disk (of total available) is used on the content node.</td>
```
Comment (vekterli) [inline en/reads-and-writes.html:251]: typo: huch -> much

File: en/reads-and-writes.html [hunk 2]
```diff
@@ -214,39 +214,58 @@ <h2 id="api-and-utilities">API and utilities</h2>
...
+  </tr>
+  <tr>
+    <th>content.proton.resource_usage.memory</th>
+    <td>A number between 0 and 1, indicating how huch memory (of total available) is used on the content node.</td>
```
Comment (vekterli) [inline en/reads-and-writes.html:255]: typo: huch -> much

File: en/reads-and-writes.html [hunk 3]
```diff
@@ -214,39 +214,58 @@ <h2 id="api-and-utilities">API and utilities</h2>
...
-java.lang.RuntimeException: ReturnCode(NO_SPACE, Put operation rejected for document 'id:test:test::0': 'diskLimitReached: {
-  action: \"add more content nodes\",
-  reason: \"disk used (0.85) &gt; disk limit (0.8)\",
-  capacity: 100000000000,
-  free: 85000000000,
-  available: 85000000000,
-  diskLimit: 0.8
-}'
+ReturnCode(NO_SPACE, External feed is blocked due to resource exhaustion:
+                     memory on node 0 [example.com] (0.82 > 0.80))
```
Comment (vekterli) [inline en/reads-and-writes.html:261]: Should hostname example be e.g. `my-vespa-node-0.example.com` to make it more obvious that it's the hostname of the actual node?

File: en/reads-and-writes.html [hunk 4]
```diff
@@ -300,16 +319,10 @@ <h2 id="feed-block">Feed block</h2>
...
-Detail resultType=FATAL_ERRORexception=
-'ReturnCode(NO_SPACE, Put operation rejected for document 'id:mynamespace:mydoc::123456' of type 'mydoc':
-'enumStoreLimitReached: {
-    action: "add more content nodes",
-    reason: "enum store address space used (0.92813) &gt; limit (0.9)",
-    enumStore: { used: 31890298144, dead: 0, limit: 34359738368},
-    attributeName: "text", subdb: "ready"}')'
-endpoint=vespa1:8080 ssl=false resultTimeLocally=1532685239428
+ReturnCode(NO_SPACE, External feed is blocked due to resource exhaustion:
+                     attribute-enum-store:test.ready.a1 on node 0 [example.com] (0.91 > 0.90))
```
Comment (vekterli) [inline en/reads-and-writes.html:325]: See other question on example hostname

**Review by vekterli — APPROVED** [review]
:+1: some minor comments/questions for consideration

File: en/reads-and-writes.html [hunk 5]
```diff
@@ -214,39 +214,58 @@ <h2 id="api-and-utilities">API and utilities</h2>
...
 <h2 id="feed-block">Feed block</h2>
 <p>
-Write operations fail when a cluster is at disk or memory capacity.
-</p><p>
+A content cluster blocks external write operations when at least one content node has reached the
+<a href="reference/services-content.html#resource-limits">resource limit</a>
+of disk or memory. This is done to avoid saturating content nodes.
+The <em>Cluster controller</em> monitors the resource usage of the content nodes
+and decides whether to block feeding.
+When feed is blocked, write operations are rejected by the <em>Distributor</em>.
```
[the remark this answers is not in the document]
Comment (geirst) [reply en/reads-and-writes.html:222]: Fixed

File: en/reads-and-writes.html [hunk 6]
```diff
@@ -214,39 +214,58 @@ <h2 id="api-and-utilities">API and utilities</h2>
...
+  <li>Update operations that are <em>not</em> <code><a href="reference/document-json-format.html#assign">assign</a></code></li>
+  <li>Update <code><a href="reference/document-json-format.html#assign">assign</a></code> operations to multi-value fields</li>
+  <li>Update <code><a href="reference/document-json-format.html#assign">assign</a></code> operations to non-numeric single-value fields</li>
+</ul>
+
+These operations are still allowed:
+<ul>
+  <li>Remove operations</li>
+  <li>Update <code><a href="reference/document-json-format.html#assign">assign</a></code> operations to numeric single-value fields</li>
+</ul>
```
[the remark this answers is not in the document]
Comment (geirst) [reply en/reads-and-writes.html:235]: Fixed

File: en/reads-and-writes.html (hunk 1 above)
Comment (geirst) [reply en/reads-and-writes.html:251]: Fixed

File: en/reads-and-writes.html (hunk 2 above)
Comment (geirst) [reply en/reads-and-writes.html:255]: Fixed

File: en/reads-and-writes.html (hunk 3 above)
Comment (geirst) [reply en/reads-and-writes.html:261]: Fixed

File: en/reads-and-writes.html (hunk 4 above)
Comment (geirst) [reply en/reads-and-writes.html:325]: Fixed

**Comment by geirst** [comment]
@vekterli PTAL

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**